matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/session/room/model/message/MessageLocationContent.kt

```diff
@@ -39,29 +39,65 @@ data class MessageLocationContent(
          */
         @Json(name = "geo_uri") val geoUri: String,
 
-        /**
-         * See https://github.com/matrix-org/matrix-doc/blob/matthew/location/proposals/3488-location.md
-         */
-        @Json(name = "org.matrix.msc3488.location") val locationInfo: LocationInfo? = null,
-
         @Json(name = "m.relates_to") override val relatesTo: RelationDefaultContent? = null,
-        @Json(name = "m.new_content") override val newContent: Content? = null,
+        @Json(name = "m.new_content") override val newContent: Content? = null
+) : MessageContent {
 
-        /**
-         * m.asset defines a generic asset that can be used for location tracking but also in other places like
-         * inventories, geofencing, checkins/checkouts etc.
-         * It should contain a mandatory namespaced type key defining what particular asset is being referred to.
-         * For the purposes of user location tracking m.self should be used in order to avoid duplicating the mxid.
-         */
-        @Json(name = "m.asset") val locationAsset: LocationAsset? = null,
+    /**
+     * See https://github.com/matrix-org/matrix-doc/blob/matthew/location/proposals/3488-location.md
+     */
+    @Json(name = "org.matrix.msc3488.location") var unstableLocationInfo: LocationInfo? = null
+    @Json(name = "m.location") var stableLocationInfo: LocationInfo? = null
 
-        /**
-         * Exact time that the data in the event refers to (milliseconds since the UNIX epoch)
-         */
-        @Json(name = "org.matrix.msc3488.ts") val ts: Long? = null,
+    @Transient
+    var locationInfo: LocationInfo? = null
+    get() = stableLocationInfo ?: unstableLocationInfo
+    set(value) {
+        field = value
+        unstableLocationInfo = value
+    }
```

Comment:
This is too complex.

```kotlin
    @Json(name = "org.matrix.msc3488.location") val unstableLocationInfo: LocationInfo? = null
    @Json(name = "m.location") val stableLocationInfo: LocationInfo? = null
```

must be val and in the constructor.

then you can have a helper

```kotlin
fun getLocationInfo() = stableLocationInfo ?: unstableLocationInfo
```

Nothing more is needed.

Reply:
Thank you, much more cleaner. We should always use unstable prefixes while MSC is not finalized.